lib/src/config_parser.cc

```diff
@@ -254,7 +285,15 @@ namespace hocon { namespace config_parser {
                         throw new bug_or_broken_exception("somehow got multi-element path in JSON mode");
                     }
 
-                    throw bug_or_broken_exception("createValueUnderPath not implemented");
+                    shared_object obj = create_value_under_path(remaining, new_value);
+                    auto existing = values.find(*key);
+                    if (existing != values.end()) {
+                        obj = dynamic_pointer_cast<const config_object>(obj->with_fallback(existing->second));
+                        assert(obj);
+                        existing->second = obj;
+                    } else {
```

Comment:
I don't believe this should be an else clause, from the looks of the Java source:

```
                        if (existing != null) {
                            obj = obj.withFallback(existing);
                        }
                        values.put(key, obj);
```